### PR TITLE
[dataset] Add a tool to generate datasets

### PR DESCRIPTION
**Pull request by yundddd**
This PR adds a tool to generate a large number of datasets. The generated data is also version tracked in datasets/ folder.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,9 +85,57 @@ affect_receive:
 
 This config injects a fault to node B at time 20 to delay its receipt of messages on `topic1`. Please see this [fault_injection.py](graph_generator/fault_injection.py) for a full list of faults supported.
 
-## Example Configs And Dataset
+## Example Configs
+
+In the [config](graph_generator/config/autonomous_vehicle/) folder, we have created an example graph that approximates autonomous vehicle software architecture. You can visualize how faults propagate with the following command:
+
+```bash
+bazel run //graph_generator:main -- --graph graph_generator/config/autonomous_vehicle/graph.yaml --node_feature_output ~/out --fault graph_generator/config/autonomous_vehicle/faults/crash_camera_driver1.yaml --fault_label_output /tmp/a --stop 200 --inject_at 10 --viz
```
Comment (alycialee) [inline README.md:93]: nit: is the extra `--` before `--graph` intentional?

File: README.md [hunk 2]
```diff
@@ -89,9 +85,57 @@ affect_receive:
...
+
+## Generating Datasets
+
+The executor is capable of generating a large number of datasets based on configurations. Users can sweep over define faults and inject them at various time. For example, by running the following command:
+
+```bash
+bazel run //graph_generator/dataset:generate_datasets -- --graph graph_generator/config/autonomous_vehicle/graph.yaml --output_dir ~/output  --fault_dir graph_generator/config/autonomous_vehicle/faults --stop 1500 --fault_begin 800 --fault_end 1100 --max_num_sweep 30
+```
+
+the executor injects all faults specified in the fault_dir directory into the graph (defined by graph.yaml), and sweeps the injection time between 800 adn 1100 time units with equal partitions (800,810,820...). Sweeping the fault injection time may allow models to learn fault transition better. The output files have the following structure:
```
Comment (alycialee) [inline README.md:106]: nit: "between 800 **and** 1100 time units"

File: README.md [hunk 3]
```diff
@@ -89,9 +85,57 @@ affect_receive:
...
+
+```
+- ~/output/
+   |_____ fault_file1_name/
+          |______ edge_index.csv
+          |______ fault_label_inject_at_800.csv
+          |______ fatul_label_inject_at_810.csc
```
Comment (alycialee) [inline README.md:113]: nit: should be csv instead of csc right?

Comment (yundddd) [reply README.md:113]:
```suggestion
          |______ fatul_label_inject_at_810.csv
```

File: README.md (hunk 2 above)
Comment (yundddd) [reply README.md:106]:
```suggestion
the executor injects all faults specified in the fault_dir directory into the graph (defined by graph.yaml), and sweeps the injection time between 800 and 1100 time units with equal partitions (800,810,820...). Sweeping the fault injection time may allow models to learn fault transition better. The output files have the following structure:
```

File: README.md (hunk 1 above)
Comment (yundddd) [reply README.md:93]: Yep. It's a way to tell bazel to pass argument to python instead of bazel itself

File: README.md [hunk 4]
```diff
@@ -89,9 +85,57 @@ affect_receive:
...
+          ...
+          |______ node_feature_inject_at_800.csv
+          |______ node_feature_inject_at_800.csv
+          ...
+   |_____ fault_file3_name/
+          |______ edge_index.csv
+          |______ fault_label_inject_at_800.csv
+          |______ fatul_label_inject_at_810.csc
```
Comment (alycialee) [inline README.md:130]: same here for "csc" - ctrl+f for `*_810.csc`

Comment (yundddd) [reply README.md:130]:
```suggestion
          |______ fatul_label_inject_at_810.csv
```

File: README.md [hunk 5]
```diff
@@ -89,9 +85,57 @@ affect_receive:
...
+          |______ fatul_label_inject_at_810.csv
+          ...
+          |______ node_feature_inject_at_800.csv
+          |______ node_feature_inject_at_800.csv
+          ...
+
+   |_____ fault_file2_name/
+          |______ edge_index.csv
+          |______ fault_label_inject_at_800.csv
+          |______ fatul_label_inject_at_810.csc
```
Comment (yundddd) [inline README.md:122]:
```suggestion
          |______ fatul_label_inject_at_810.csv
```

File: graph_generator/main.py [hunk 6]
```diff
@@ -114,5 +74,74 @@ def main(
...
+    graph: str,
+    fault: str | None,
+    viz: bool,
+    stop: int,
+    edge_index_output: str,
+    node_feature_output: str | None,
+    fault_label_output: str | None,
+    inject_at: int | None,
+):
+    """
```
Comment (yundddd) [inline graph_generator/main.py:127]:
```suggestion
    
```

File: graph_generator/main.py [hunk 7]
```diff
@@ -114,5 +74,74 @@ def main(
...
+      A
```
Comment (yundddd) [inline graph_generator/main.py:128]:
```suggestion
      
```

File: graph_generator/main.py [hunk 8]
```diff
@@ -114,5 +74,74 @@ def main(
...
+     / \
```
Comment (yundddd) [inline graph_generator/main.py:129]:
```suggestion
```

File: graph_generator/main.py [hunk 9]
```diff
@@ -114,5 +74,74 @@ def main(
...
+    B   C
```
Comment (yundddd) [inline graph_generator/main.py:130]:
```suggestion
```

File: graph_generator/main.py [hunk 10]
```diff
@@ -114,5 +74,74 @@ def main(
...
+   /     \
```
Comment (yundddd) [inline graph_generator/main.py:131]:
```suggestion
```

File: graph_generator/main.py [hunk 11]
```diff
@@ -114,5 +74,74 @@ def main(
...
+  D       E
```
Comment (yundddd) [inline graph_generator/main.py:132]:
```suggestion
```

File: graph_generator/main.py [hunk 12]
```diff
@@ -114,5 +74,74 @@ def main(
...
+    """
```
Comment (yundddd) [inline graph_generator/main.py:133]:
```suggestion
```

**Review by alycialee — APPROVED** [review]

**Pull request merged**